src/pages/home/ReportScreen.js

```diff
@@ -160,12 +156,29 @@ function ReportScreen({
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const {isSmallScreenWidth} = useWindowDimensions();
-
-    const firstRenderRef = useRef(true);
+    const {isOffline} = useNetwork();
     const flatListRef = useRef();
     const reactionListRef = useRef();
+    const firstRenderRef = useRef(true);
+    const reportIDFromRoute = getReportID(route);
+    const reportActionIDFromRoute = lodashGet(route, 'params.reportActionID', null);
+    const shouldTriggerLoadingRef = useRef(!!reportActionIDFromRoute);
     const prevReport = usePrevious(report);
     const prevUserLeavingStatus = usePrevious(userLeavingStatus);
+    const [isPrepareLinkingToMessage, setLinkingToMessage] = useState(!!reportActionIDFromRoute);
+
+    const reportActions = useMemo(() => {
+        if (!!allReportActions && allReportActions.length === 0) {
+            return [];
+        }
+        const sortedReportActions = ReportActionsUtils.getSortedReportActionsForDisplay(allReportActions, false);
+        const currentRangeOfReportActions = ReportActionsUtils.getContinuousReportActionChain(sortedReportActions, reportActionIDFromRoute);
+        // eslint-disable-next-line rulesdir/prefer-underscore-method
+        const reportActionsWithoutDeleted = currentRangeOfReportActions.filter((item) => ReportActionsUtils.shouldReportActionBeVisible(item, item.reportActionID));
+        return reportActionsWithoutDeleted;
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [reportActionIDFromRoute, allReportActions, isOffline]);
```

Comment:
Why does `isOffline` have to be in the dependency array here?

Reply:
It came from here https://github.com/Expensify/App/pull/32474/files#diff-dd4bfa50713397aca8cb40145317936ab0affe19abc2a9c24d6c8849ed75dc9bR174
Now I see that it was removed from the main, and it looks like my changes now. I'll remove it.